contracts/automation/andromeda-condition/src/contract.rs

```diff
@@ -0,0 +1,1054 @@
+    let res = RESULTS.load(deps.storage)?;
+    ensure!(!res.is_empty(), ContractError::NoResults {});
+
+    let contract_addr =
+        EXECUTE_ADO
+            .load(deps.storage)?
+            .get_address(deps.api, &deps.querier, app_contract)?;
+    match logic {
+        LogicGate::AND =>
+        // We don't want to find a false bool, so we want it to return false
+        {
+            // At least two results should be available
+            ensure!(res.len() > 1_usize, ContractError::NotEnoughResults {});
+
+            ensure!(
+                !res.iter().any(|x| x == &false),
+                ContractError::UnmetCondition {}
+            );
+
+            // Reset stored results after they meet our conditions
+            let new: Vec<bool> = vec![];
+            RESULTS.save(deps.storage, &new)?;
+
+            Ok(Response::new()
+                .add_message(CosmosMsg::Wasm(WasmMsg::Execute {
+                    contract_addr,
+                    msg: to_binary(&execute::ExecuteMsg::Execute {})?,
+                    funds: vec![],
+                }))
+                .add_attribute("result", "sent by AND".to_string()))
+        }
+        // Just one result being true meets our condition
+        LogicGate::OR => {
+            ensure!(
+                res.iter().any(|x| x == &true),
+                ContractError::UnmetCondition {}
+            );
+
+            // Reset stored results after they meet our conditions
+            let new: Vec<bool> = vec![];
+            RESULTS.save(deps.storage, &new)?;
+
+            Ok(Response::new()
+                .add_message(CosmosMsg::Wasm(WasmMsg::Execute {
+                    contract_addr,
+                    msg: to_binary(&execute::ExecuteMsg::Execute {})?,
+                    funds: vec![],
+                }))
+                .add_attribute("result", "sent by OR".to_string()))
+        }
+        // At least one result should be true, but not all of them
+        LogicGate::XOR => {
+            // At least two results should be available
+            ensure!(res.len() > 1_usize, ContractError::NotEnoughResults {});
+
+            ensure!(
+                !res.iter()
+                    .all(|x| x == &true && res.iter().any(|x| x == &true)),
+                ContractError::UnmetCondition {}
+            );
+
+            // Reset stored results after they meet our conditions
+            let new: Vec<bool> = vec![];
+            RESULTS.save(deps.storage, &new)?;
+
+            Ok(Response::new()
+                .add_message(CosmosMsg::Wasm(WasmMsg::Execute {
+                    contract_addr,
+                    msg: to_binary(&execute::ExecuteMsg::Execute {})?,
+                    funds: vec![],
+                }))
+                .add_attribute("result", "sent by XOR".to_string()))
+        }
+        // Only takes one input, takes false as true
+        LogicGate::NOT => {
+            ensure!(res.len() == 1 && !res[0], ContractError::UnmetCondition {});
```

Comment:
We need validation on these conditions are instantiation I think. Maybe a validation function on the `InstantiateMsg`